### PR TITLE
LPS-153350 Add validation to groovy script maximum number of lines available

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-153350

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 1]
```diff
@@ -277,6 +279,11 @@ else if (Objects.equals(
 						engine,
 						ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY)) {
 
+				if (StringUtil.count(script, StringPool.NEW_LINE) > 2987) {
+					throw new ObjectValidationRuleScriptException(
+						"the-maximum-number-of-lines-available-is-2987");
+				}
+
```
Comment (gabrielwas) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:286]: @danvb112 can you put the catch like this?

```
catch (Exception exception) {
     if (_log.isDebugEnabled()) {
        _log.debug(exception);
    }

    if(exception instanceof ObjectValidationRuleScriptException){
        throw exception;
    }

    throw new ObjectValidationRuleScriptException("syntax-error");
}
```

could you test with the API call and check if the right exception is thrown? Thx.

**Review by gabrielwas — CHANGES_REQUESTED** [review]

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by gabrielwas** [comment]
@danvb112 Have you checked the API if we're returning the right exceptions?

**Comment by danvb112** [comment]
Hey @gabrielwas, yes I checked the api return and it is working, the return is:

Screenshot from 2022-07-28 12-39-16

**Comment by gabrielwas** [comment]
@danvb112 perfect! Thank you!

**Review by gabrielwas — APPROVED** [review]

**Comment by gabrielwas** [comment]
ci:forward

**Pull request closed without merging**